src/components/Table/GridTable.tsx

```diff
@@ -345,29 +331,44 @@ export function GridTable<R extends Kinded, S = {}, X extends Only<GridTableXss,
     const headerRows: RowTuple<R>[] = [];
     const filteredRows: RowTuple<R>[] = [];
 
+    // Misc state to track our nested card-ification, i.e. interleaved actual rows + chrome rows
+    const nestedCards = !!style.nestedCards && new NestedCards(columns, filteredRows, style);
+
     // Depth-first to filter
     function visit(row: GridDataRow<R>): void {
-      if (row.kind === "header") {
-        headerRows.push([row, makeRowComponent(row)]);
-        return;
-      }
-
       const passesFilter =
         filters.length === 0 ||
         filters.every((filter) =>
           columns.map((c) => applyRowFn(c, row)).some((maybeContent) => matchesFilter(maybeContent, filter)),
         );
+      // Even if we don't pass the filter, one of our children might, so we continue on after this check
       if (passesFilter) {
+        nestedCards && nestedCards.beginRow(row);
         filteredRows.push([row, makeRowComponent(row)]);
       }
 
       const isCollapsed = collapsedIds.includes(row.id);
       if (!isCollapsed && row.children) {
-        row.children.forEach(visit);
+        visitRows(row.children);
       }
+
+      nestedCards && nestedCards.endRow();
     }
 
-    maybeSorted.forEach(visit);
```

Comment:
Instead of just `maybeSorted.forEach(visit)` we call `visitRows` to kick things off b/c that has the logic to add spacers between cards.